packages/react-ui/src/widgets/legend/LegendRamp.js

```diff
@@ -20,13 +20,22 @@ function LegendRamp({ isContinuous = false, legend }) {
     !labels.length || isContinuous ? 2 : labels.length + 1
   );
 
-  let [min, max] = getMinMax({ labels });
+  // If labels is complex, its type is { value: number, label: string }[]
+  const isComplexLabels = !!labels[0]?.label;
+
+  const values = isComplexLabels ? labels.map(({ value }) => value) : labels;
+  const formattedLabels = isComplexLabels ? labels.map(({ label }) => label) : labels;
+
+  const [min, max] = getMinMax({ labels: values });
 
   const error = Number.isNaN(min) || Number.isNaN(max);
 
+  let maxLabel = formattedLabels[formattedLabels.length - 1];
+  let minLabel = formattedLabels[0];
+
   if (!isContinuous) {
-    min = '< ' + min;
-    max = '≥ ' + max;
+    minLabel = '< ' + minLabel;
```

Comment:
As you said @bbecquet, this isn't the goal of the PR, so for further comment I suggest you to discuss this in a RFC, so we don't block this PR.

In any case, the explanation of why is this the behavior is:

[CARTO colorBins methods](https://github.com/visgl/deck.gl/blob/master/modules/carto/src/style/color-bins-style.ts) uses [d3.scaleThreshold](https://github.com/d3/d3-scale/blob/main/README.md#scaleThreshold), so it follows its behavior that is well explained both in d3 documentation like in [the deck.gl one](https://deck.gl/docs/api-reference/carto/styles#colorbins).
When using the discrete ramp, you indicate the bins which in most case do not include the min/max as one of their bins. So, the first value we consider in the legend is not the min of the dataset, so more values are on the left of it; and the same happens with the "max".

Using these bins: `[1e5, 2e5, 3e5]`

The ranges we consider are:
```
[, 1e5)
[1e5, 2e5)
[2e5, 3e5)
[3e5,]
```

Reply:
@Clebal Thanks a lot for the explanation and the doc pointers. I guess I have to dive a bit more into this to understand how it could match our usage in builder legends.